metadata-ingestion/src/datahub/ingestion/source/aws/dynamodb.py

```diff
@@ -0,0 +1,188 @@
+import logging
+from collections import OrderedDict
+from typing import Dict, Iterable, List
+from dataclasses import dataclass, field
+
+from datahub.configuration.source_common import PlatformSourceConfigBase
+from datahub.emitter.mce_builder import (
+    make_data_platform_urn,
+    make_dataset_urn,
+)
+from datahub.emitter.mcp import MetadataChangeProposalWrapper
+from datahub.ingestion.api.common import PipelineContext
+from datahub.ingestion.api.source import Source, SourceReport
+from datahub.ingestion.api.workunit import MetadataWorkUnit
+from datahub.ingestion.source.aws.aws_common import AwsSourceConfig
+from datahub.metadata.schema_classes import (
+    ChangeTypeClass,
+)
+from datahub.metadata.com.linkedin.pegasus2avro.schema import (
+    BooleanType,
+    BytesType,
+    KeyValueSchema,
+    NumberType,
+    SchemaField,
+    SchemaFieldDataType,
+    SchemaMetadata,
+    StringType,
+)
+
+
+logger = logging.getLogger(__name__)
+
+DEFAULT_PLATFORM = "dynamodb"
+
+_attribute_type_mapping = {
+    "S": StringType,
+    "N": NumberType,
+    "B": BytesType,
+    "BOOL": BooleanType,
+}
+
+
+class DynamoDBSourceConfig(AwsSourceConfig, PlatformSourceConfigBase):
+
+    @property
+    def dynamodb_client(self):
+        return self.get_dynamodb_client()
+
+    @property
+    def dynamodbstreams_client(self):
+        return self.get_dynamodbstreams_client()
+
+
+@dataclass
+class DynamoDBSourceReport(SourceReport):
+    filtered_tables: List[str] = field(default_factory=list)
+    streaming_disabled_tables: List[str] = field(default_factory=list)
+
+    def report_table_filtered(self, table: str) -> None:
+        self.filtered_tables.append(table)
+
+    def report_table_streaming_disabled(self, table: str) -> None:
+        self.streaming_disabled_tables.append(table)
```

Comment:
Should these be annotated with `@classmethod` given that they are updating class-level attributes?